src/stores/events/EventsStore.ts

```diff
@@ -402,6 +380,47 @@ export default class EventsStore {
 		return this.selectedPath.some(n => n.eventId === eventTreeNode.eventId);
 	}
 
+	private init = async (defaultState: EventStoreDefaultStateType) => {
+		let initialState = !defaultState || typeof defaultState === 'string' ? {} : defaultState;
+
+		this.filterStore = new EventsFilterStore(this.graphStore, this.searchPanelStore, {
+			filter: initialState?.filter,
+			range: initialState?.range,
+		});
+		this.viewStore = new ViewStore({
+			flattenedListView: initialState.flattenedListView,
+			panelArea: initialState.panelArea,
+		});
+		this.searchStore = new EventsSearchStore(this.api, this, {
+			searchPatterns: initialState.search,
+		});
+		this.eventDataStore = new EventsDataStore(this, this.filterStore, this.api);
+
+		if (typeof defaultState === 'string') {
+			const event = await this.api.events.getEvent(defaultState);
+			this.filterStore.setRange(
+				getRangeFromTimestamp(timestampToNumber(event.startTimestamp), this.graphStore.interval),
+			);
+			initialState = { ...initialState, selectedEventId: event.eventId, targetEvent: event };
+			this.goToTargetEvent(initialState);
+			return;
+		}
+
+		this.goToTargetEvent(initialState);
+	};
+
+	private goToTargetEvent = (defaultState: EventStoreDefaultState) => {
```

Comment:
this function just calls other methods, why don't we just call `goToEvent` or `fetchEventTree` inside init?